### PR TITLE
Disable ActionController auto params wrapping

### PR DESCRIPTION
**Pull request by rhymes**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Rails by default accepts wrapped and unwrapped parameters in JSON requests (with or without a root object), regardless of the fact that the contract with the client requests wrapped parameters or not. This results in unforeseeable bugs when the client sends unwrapped parameters in lieu of wrapped ones.

As undeterministic bugs are a PITA and this is the classic grey area in Rails apps and as the customer facing API is fully documented and the internal can be manually adjusted, we disable this and be done with this odd feature.

## Related Tickets & Documents

Closes https://github.com/forem/forem/issues/11645
https://github.com/forem/forem/pull/11638

## QA Instructions, Screenshots, Recordings

1. send unwrapped parameters to `POST localhost:3000/api/articles`

```json
{
        "title": "A test post again",
        "description": "Testing article creation bug again",
        "body_markdown": "## Some content",
        "tags": ["javascript"]
}
```

1. see it fail loudly instead of work silently but ignoring tags

## Added tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## Added to documentation?

- [ ] [Developer Docs](https://docs.forem.com) and/or [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [ ] README
- [x] No documentation needed

### CODE REVIEW

**Review by Ridhwana — APPROVED** [review]

File: body.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (citizen428) [inline body.json:1]: Should this file be here at the project root directory?

File: config/initializers/wrap_parameters.rb [hunk 2]
```diff
@@ -5,7 +5,9 @@
 
 # Enable parameter wrapping for JSON. You can disable this by setting :format to an empty array.
 ActiveSupport.on_load(:action_controller) do
-  wrap_parameters format: [:json]
+  # Globally disable auto wrapping resulting in a stricter API and to avoid bugs with array payloads.
```
Comment (citizen428) [inline config/initializers/wrap_parameters.rb:8]: Good comment/context, 🔥

File: body.json (hunk 1 above)
Comment (rhymes) [reply body.json:1]: AHHAHA omg this should definitely not be here. Thanks for noticing

**Review by mstruve — APPROVED** [review]
LGTM!

**Pull request merged**